app/components/portfolios/details_component.rb

```diff
@@ -54,10 +54,39 @@ def all_subprojects
       all_descendants.project
     end
 
+    def render_sub_status_bar?
+      # When none of the descendants have a status set, there's nothing to show and we
+      # will return false.
+      sub_statuses.keys.any?(&:present?)
+    end
+
+    def sub_statuses_with_percentages
+      return @status_percentage if @status_percentage
+
+      statuses = sub_statuses
+      total = statuses.values.sum
+
+      @status_percentage = statuses.map do |code, count|
+        percentage = ((count.to_f / total) * 100).round
+
+        { code:, count:, percentage: }
+      end
+    end
+
+    def sub_status_hover_card_id
+      "portfolio-progress-hover-card-#{portfolio.id}"
+    end
+
     private
 
     def all_descendants
       @all_descendants ||= portfolio.descendants.visible
     end
+
+    def sub_statuses
+      @sub_statuses ||= all_descendants.where("workspace_type = ? OR workspace_type = ?", "project", "program")
```

Comment:
Already discussed in chat - the condition is not needed, as sub-portfolios are allowed